src/HealthChecks.RabbitMQ/README.md

```diff
@@ -0,0 +1,130 @@
+# RabbitMQ Health Check
+
+This health check verifies the ability to communicate with a RabbitMQ server
+
+## Example Usage
+
+With all of the following examples, you can additionally add the following parameters:
+
+- `name`: The health check name. Default if not specified is `rabbitmq`.
+- `failureStatus`: The `HealthStatus` that should be reported when the health check fails. Default is `HealthStatus.Unhealthy`.
+- `tags`: A list of tags that can be used to filter sets of health checks.
+- `timeout`: A `System.TimeSpan` representing the timeout of the check.
+
+### Basic
+
+This will create a new `IConnectionFactory` and `IConnection` on every request to get the health check result. Use
+the extension method where you provide the `Uri` to connect with. You can optionally set the `SslOption` if needed.
+
+```cs
+public void ConfigureServices(IServiceCollection services)
+{
+    services
+        .AddHealthChecks()
+        .AddRabbitMQ("amqps://user:pass@host/vhost");
+}
+```
+
+### Dependency Injected `IConnection`
+
+As per [RabbitMQ docs](https://www.rabbitmq.com/connections.html) and its suggestions on
+[high connectivity churn](https://www.rabbitmq.com/networking.html#dealing-with-high-connection-churn), connections are meant to be long lived.
+Ideally, this should be configured as a singleton.
+If you are sharing a single connection for every time a health check is requested,
+you must ensure automatic recovery is enable so that the connection can be re-established if lost.
+
+```cs
+public void ConfigureServices(IServiceCollection services)
+{
+    var factory = new ConnectionFactory()
+    {
+        Uri = new Uri("amqps://user:pass@host/vhost"),
+        AutomaticRecoveryEnabled = true
+    };
+
+    var connection = factory.CreateConnection();
```

Comment:
I know this is only some sample code, but people frequently copy/paste these things and assume the code fit for production. In this case, I don't think we want to be creating a real connection in `ConfigureServices`. We want our health check to fail, and as written I think the app will fail to start.

I think we can keep this connection lazy without too much hassle using another `AddSingleton` overload:

```csharp
.AddSingleton<IConnection>(_ => factory.CreateConnection())
```